octopod-backend/src/Octopod/Server/ControlScriptUtils.hs

```diff
@@ -194,24 +193,26 @@ runCommandArgs f args = do
   runCommandArgs' cmd args
 
 runCommandArgs' ::
-  (MonadBase IO m, HasType TimedFastLogger r, MonadReader r m) =>
+  (KatipContext m, MonadBase IO m) =>
   Command ->
   ControlScriptArgs ->
   m (ExitCode, Stdout, Stderr)
 runCommandArgs' (Command cmd) (ControlScriptArgs args) = do
-  logger <- asks (getTyped @TimedFastLogger)
-  let logText = T.unwords (cmd : fmap T.pack args)
-  liftBase $ logInfo logger $ "calling: " <> logText
-  res@(ec, _, _) <- runCommand (T.unpack cmd) args
-  liftBase $
-    logInfo logger $
-      "calling `" <> logText <> "` exited with: " <> show' ec
-  return res
+  runCommand (T.unpack cmd) args
 
 -- | Helper to run command with pipes.
-runCommand :: MonadBase IO m => FilePath -> [String] -> m (ExitCode, Stdout, Stderr)
+runCommand :: (KatipContext m, MonadBase IO m) => FilePath -> [String] -> m (ExitCode, Stdout, Stderr)
 runCommand cmd args = do
+  logLocM DebugS $ "Calling control script: " <> logStr cmd <> " " <> show' args
   (ec, out, err) <- liftBase . readProcess $ proc cmd args
+  logLocM (if ec /= ExitSuccess then WarningS else DebugS) $
+    "Control script " <> logStr cmd <> " " <> show' args
+      <> " exited with: "
+      <> show' ec
+      <> "\nStdout was:\n"
+      <> logStr out
+      <> "\nStderr was:\n"
+      <> logStr err
```

Comment:
The output is not uniform in the way whitespace is distributed (due to the specifics of what control scripts output):
<img width="676" alt="Screenshot 2021-09-22 at 01 10 03" src="https://user-images.githubusercontent.com/6209627/134254898-6ea9cf79-7f37-4493-9cd4-5d2a14725752.png">

I propose we do the following:
1. [ ] If some output handle of the script wasn't written to (length is zero), then don't output it at all.

When running the server _locally_ (currently `--minimal`):

2. [ ] It would be nice to indent the output when running the server _locally_ . (`unlines . fmap ("\t" <>) . lines`)
3. [ ] Output the following elements on separate lines (easier to visually parse): command itself, exit code, stdout, stderr, so that it would look something like this:
```
script:    foo.sh --foo --bar
exit code: ExitSuccess
stderr:
    foo bar baz
    qux
stdout:
    foo bar
    bar buz qux
```

Reply:
When not in "local development mode" logging of a single item (a control script call) should **not** span multiple lines since most logging aggregation tools split log entries by newlines. Control script calling results should probably be recorded as structure fields (not just log text) and interpreted properly when "pretty-printing the entries" (see `formatter` from `octopod-backend/src/Octopod/Server/Logging.hs`)

---

On second thought the multi-line splitting might be solved by JSON logging, but the structural logging point still stands.